app/controllers/vaccines_controller.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+# VaccinesController: patient immunizations
+class VaccinesController < ApplicationController
+  include VaccineQueryHelper
+
+  before_action :authenticate_user!
+
+  def index
+    redirect_to(root_url) && return unless current_user&.can_view_patient_vaccines?
+
+    # Validate params and handle errors if invalid
+    begin
+      data = validate_table_query(params)
```

Comment:
can we rename this to `validate_vaccines_query` since the one for patients is `validate_patients_query`?

Reply:
Yep, done here: https://github.com/SaraAlert/SaraAlert/pull/721/commits/66e9a6d482854054a97a2f0ac8c1ec4c0c6af3ca#diff-cdf69598826e24a451843fb600155d000e343d9d9ad213121948b258135c9f99R6